### DISCUSSION TITLE
[New Feature] Make emitted light depend on power

### DISCUSSION THREAD

**Discussion by Crimsonum** [Vinifera-Developers/Vinifera#1333] [Feature Requests]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I have attached as much information as possible *(screenshots, videos, links, etc)*.

### Description

Light emitted by buildings (e.g. `[GALITE]`) is not disabled when the building runs out of power, even if the building has negative `Power` and/or `Powered` enabled.

### Proposed Code (optional)

If the light-emitting building drains power (has negative `Power` value) and has `Powered` enabled, the light should be disabled if the building runs out of power.

None of the light-emitting buildings in the vanilla game consume power, so it should be safe to make this the new default behavior.

**Comment by djohe** [comment]
I also would like the Laser Fence Sections and Firestorm Wall Sections to turn off light-emission together with the laser wall or firestorm wall state or when under the influence of cloaking device/field if given light-emission.
AlphaLights are completly unaffected by power state even if negative power drain and Powered=true and I think they should be affected by power state or when under the influence of cloaking device/field.
![TibSun_light_vs_power](https://user-images.githubusercontent.com/771748/227593539-85468bf4-844d-44b4-97bf-12ab4897c519.png)
TOP: Everything has enough power. MIDDLE: Everything toggled off (except firestorm) BOTTOM: Everything low power
